src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.iOS/Airship.cs

```diff
@@ -304,34 +304,35 @@ public void AddCustomEvent(CustomEvent customEvent)
 
             if (customEvent.PropertyList != null)
             {
-                NSDictionary<NSString, NSObject> propertyDictionary = new NSDictionary<NSString, NSObject>();
-                foreach (var property in customEvent.PropertyList)
+                NSMutableDictionary<NSString, NSObject> propertyDictionary = new NSMutableDictionary<NSString, NSObject>();
+                foreach (dynamic property in customEvent.PropertyList)
                 {
                     if (string.IsNullOrEmpty(property.name))
                     {
                         continue;
                     }
 
+                    NSString key = (NSString)property.name;
                     if (property is CustomEvent.Property<string> stringProperty)
                     {
-                        propertyDictionary.SetValueForKey((NSString)stringProperty.value, (NSString)stringProperty.name);
+                        propertyDictionary.SetValueForKey((NSString)stringProperty.value, key);
                     }
                     else if (property is CustomEvent.Property<double> doubleProperty)
                     {
-                        propertyDictionary.SetValueForKey((NSNumber)doubleProperty.value, (NSString)doubleProperty.name);
+                        propertyDictionary.SetValueForKey((NSNumber)doubleProperty.value, key);
                     }
                     else if (property is CustomEvent.Property<bool> boolProperty)
                     {
-                        propertyDictionary.SetValueForKey((NSNumber)boolProperty.value, (NSString)boolProperty.name);
+                        propertyDictionary.SetValueForKey((NSNumber)boolProperty.value, key);
                     }
                     else if (property is CustomEvent.Property<string[]> stringArrayProperty)
```

Comment:
We can accept any JSOn value now, do we really need to iterate through?

Reply:
I think we can factorised by casting it to NSObject directly but It should be working with string, bool and double but still need to cast it for the array:

```
 NSString key = (NSString)property.name;
 NSObject value = NSObject.FromObject(property.value);

 if (property is CustomEvent.Property<string[]> stringArrayProperty)
 {
    value = NSArray.FromObjects(stringArrayProperty.value);
    propertyDictionary.SetValueForKey(value, key);
 }
 if (value != null)
 {
     propertyDictionary.SetValueForKey(value, key);
 }
```

What do you think ?